libsql-server/src/hrana/http/stream.rs

```diff
@@ -224,6 +224,10 @@ impl<'srv, D: Connection> Guard<'srv, D> {
             None
         }
     }
+
+    pub fn closed(&self) -> bool {
+        self.stream.is_none() || self.stream.as_ref().unwrap().db.is_none()
```

Comment:
nitpick: the comment in Guard says that `self.stream` is set to `None` only in the destructor - so perhaps it's safer to unwrap it unconditionally and panic if we broke that assumption? All the other functions seem to do that

Reply:
/cc @MarinPostma